core/src/main/java/org/javarosa/core/model/instance/TreeReferenceLevel.java

```diff
@@ -141,8 +141,8 @@ public boolean equals(Object o) {
         }
 
         // predicates match-up
-        if ((predicates == null && l.predicates != null) ||
-                (l.predicates == null && predicates != null) ||
+        if (predicates == null ||
+                l.predicates == null ||
```

Comment:
Remove check to ensure both predicates aren't `null`, since that is done right above on line 139.